posthog/temporal/data_imports/workflow_activities/import_data.py

```diff
@@ -216,6 +216,37 @@ async def import_data_activity(inputs: ImportDataActivityInputs):
             else None,
         )
 
+        return await _run(
+            job_inputs=job_inputs,
+            source=source,
+            logger=logger,
+            inputs=inputs,
+            schema=schema,
+            reset_pipeline=reset_pipeline,
+        )
+    elif model.pipeline.source_type == ExternalDataSource.Type.SALESFORCE:
+        from posthog.temporal.data_imports.pipelines.salesforce.auth import salesforce_refresh_access_token
+        from posthog.temporal.data_imports.pipelines.salesforce import salesforce_source
+
+        subdomain = model.pipeline.job_inputs.get("salesforce_subdomain")
+        salesforce_access_token = model.pipeline.job_inputs.get("salesforce_access_token", None)
+        refresh_token = model.pipeline.job_inputs.get("salesforce_refresh_token", None)
+        if not refresh_token:
+            raise ValueError(f"Salesforce refresh token not found for job {model.id}")
+
+        if not salesforce_access_token:
+            salesforce_access_token = salesforce_refresh_access_token(refresh_token)
```

Comment:
We likely will need to do this on every run, not only when the access token doesn't exist